### PR TITLE
JAX-RS: add code.namespace and code.function attributes

### PR DESCRIPTION
**Pull request by laurit**

### CODE REVIEW

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/JaxRsAnnotationsTracer.java [hunk 1]
```diff
@@ -51,12 +52,16 @@ public Context startSpan(Context parentContext, Class<?> target, Method method)
     Span span = spanBuilder(parentContext, "jax-rs.request", INTERNAL).startSpan();
     updateSpanNames(
         parentContext, span, ServerSpan.fromContextOrNull(parentContext), target, method);
+    setCodeAttributes(span, target, method);
```
Comment (trask) [inline instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/JaxRsAnnotationsTracer.java:55]: if possible, better to set attributes on the SpanBuilder above

File: instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/JaxRsAnnotationsTracer.java [hunk 2]
```diff
@@ -51,12 +52,16 @@ public Context startSpan(Context parentContext, Class<?> target, Method method)
...
     return parentContext.with(span);
   }
 
   public void updateSpanNames(
       Context context, Span span, Span serverSpan, Class<?> target, Method method) {
-    String pathBasedSpanName = ServletContextPath.prepend(context, getPathSpanName(target, method));
+    String pathBasedSpanName = getPathSpanName(target, method);
+    if (pathBasedSpanName != null) {
```
Comment (trask) [inline instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/JaxRsAnnotationsTracer.java:62]: I'm not seeing how `pathBasedSpanName` can be null here?

Comment (laurit) [reply instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/JaxRsAnnotationsTracer.java]: Thanks, that was careless of me.

File: instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/JaxRsAnnotationsTracer.java [hunk 3]
```diff
@@ -48,15 +50,24 @@ public Context startSpan(Context parentContext, Class<?> target, Method method)
...
   }
 
   public void updateSpanNames(
       Context context, Span span, Span serverSpan, Class<?> target, Method method) {
-    String pathBasedSpanName = ServletContextPath.prepend(context, getPathSpanName(target, method));
+    String pathBasedSpanName = getPathSpanName(target, method);
+    // If path based name is empty skip appending context path so that path based name would
+    // remain as an empty string for which we skip updating span name. Path base span name is
+    // empty when method and class don't have a jax-rs path annotation, this can happen when
+    // creating an "abort" span, see RequestContextHelper.
```
Comment (trask) [inline instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/JaxRsAnnotationsTracer.java:67]: ❤️

**Review by trask — APPROVED** [review]

**Pull request merged**